test/test_jsinterp_parser.py

```diff
@@ -505,29 +505,114 @@ def test_precedence(self):
         ]
         self.assertEqual(list(traverse(list(jsi.statements()))), list(traverse(ast)))
 
-    @unittest.skip('Parsing function declaration not yet implemented')
     def test_call(self):
         jsi = JSInterpreter('''
         function x() { return 2; }
         function y(a) { return x() + a; }
         function z() { return y(3); }
         ''')
 
-        ast = []
+        ast = [
+            (Token.FUNC, 'x',
+             [],
+             (Token.BLOCK, [
+                 (Token.RETURN, (Token.EXPR, [
+                     (Token.ASSIGN, None, (Token.OPEXPR, [(Token.MEMBER, (Token.INT, 2), None, None)]), None)
+                 ])
+                  )
+             ])),
+            (Token.FUNC, 'y',
+             ['a'],
+             (Token.BLOCK, [
+                 (Token.RETURN, (Token.EXPR, [
+                     (Token.ASSIGN, None,
+                      (Token.OPEXPR, [
+                          # Not sure about this one
+                          (Token.MEMBER, (Token.ID, 'x'), None, (Token.CALL, [], None)),
+                          (Token.MEMBER, (Token.ID, 'a'), None, None),
+                          (Token.OP, _OPERATORS['+'][1])
+                      ]), None)
+                 ])
+                  )
+             ])),
+            (Token.FUNC, 'z',
+             [],
+             (Token.BLOCK, [
+                 (Token.RETURN, (Token.EXPR, [
+                     (Token.ASSIGN, None, (Token.OPEXPR, [
+                         # Not sure about this one
+                         (Token.MEMBER, (Token.ID, 'y'), None, (Token.CALL, [
```

Comment:
If I'm not mistaken args is not **None** if MemberExpression is a NewExpresion. Therefore this is correct.